freezeyt_blog/articles/meeting30.md

```diff
@@ -0,0 +1,93 @@
+
+Potom jsme se dozvěděli, jak pokračovat s našimi rozpracovanými pull requesty.
+
+Následně jsme se vrátili k nastavení serveru WSGI a hledali jsme,
+jaké všechny hlavičky musíme předat dle standardu.
+Také jsme pokračovali vyhazováním příslušných chyb.
+Vypadá to, že vše z poznámek, které jsme do issue napsali před několika týdny,
+je zapracováno.
+
+Takže jsme se vrátili k rozpracovanému pull requestu.
+V průběhu commitování jsme objevili necommitnutou změnu,
+která přidávala `SERVER_SOFTWARE` s verzí freezeyt do slovníku `environ`.
+Následně jsme uvažovali, proč jsme to tam chtěli přidat.
+Tak jsme opravili změny, na které si stěžovaly pyflakes.
+
+Koukli jsme se na otevřené issues a rozhodli jsme se pracovat na tom,
+že freezeyt smaže existující adresář, před tím, než do něj začne ukládat stránku.
+
+Začali jsme, jako už klasicky dle TDD, napsáním testů.
+A pak jsme je začali opravovat.
+Poté, co jsme začali upravovat freezeyt jsme zjistili,
+že vestavěná chyba `FileExistsError` není úplně výstižná.
+Výjimky v Pythonu jsou v jisté [hierarchii](https://docs.python.org/3/library/exceptions.html#exception-hierarchy).
+Přesnější by bylo, kdyby se chyba jmenovala `DirectoryExistsError`.
+A tak jsme si ji vytvořili.
+Vytvoření vlastní chyby v Pythonu znamená podědit z nějaké třídy
+z výše zmiňované hierarchie.
+Následně jsme přidali možnost smazat původní adresář.
+PyFlakes si stěžovalo jen na to, že u některých formátovacích řetězců
+nevkládáme žádnou proměnnou.
+Do README jsme napsali poznámku o tom, že freezeyt smaže existující adresář.
+
+> Tip pro práci v terminálu, když píšete nějaký příkaz, ale ještě přesně nevíte,
+> jak by měl končit, na začátek řádku napište `#` a tím to „zakomentujete“.
+> (Ekvivalentní příkaz na Windows je `rem`.)
+
+Rozhodli jsme se z repozitáře smazat soubory s „divnými“ názvy.
+Tím jsme vyřešili jedno issue.
+Abychom nepřišli o testy, které tyto soubory používaly,
+převedli jsme je na testy pouze s výstupem do slovníku.
+
+Nakonec jsme zkontrolovali, že v repozitáři nezbyly žádné soubory s divnými jmény.
+> Na to jsme použili tento příkaz: `ls -R | grep '[^-a-zA-Z0-9._/:]'`.
+> Tento příkaz nebude fungovat na Windows.
+> `ls -R` - rekurzivně prohledá adresář (a vrátí názvy souborů).
+> Pomocí `|` předáme výstup z předchozího programu dalšímu programu.
+> V tomto případě je to `grep`, který najde výsledky,
+> které odpovídají danému regulárnímu výrazu.
+> regulární výraz `[^-a-zA-Z0-9._/:]` odpovídá všemu bez pomlčky, tečky,
+> podtržítka, lomítka, dvojtečky velkých a malých písmen (bez diakritiky)
+> a číslic v názvu souboru.
+
+Na chvíli jsme se zastavili u příkazu `git cherry-pick` - ten může na danou
+větev přidat i commit, ke kterému se nedostaneme z žádné větve
+(protože byla například smazána).
+Dostaneme se k němu právě proto, že známe jeho commit hash.
+V Gitu existuje [příkaz](), který by tyto commity ze „vzduchoprázdna“ dokáže vylovit.
```

Comment:
I am surprised, that GH Actions didn't have a problem with the incomplete link, last time I tried to do this in a freezeyt blog article it ended with an error saying it found a broken link.

Reply:
```suggestion
V Gitu existuje [příkaz](https://git-scm.com/docs/git-fsck#Documentation/git-fsck.txt---no-reflogs), který by tyto commity ze „vzduchoprázdna“ dokáže vylovit.
```

Reply:
The command is `git fsck`.